core/sched.c

```diff
@@ -221,3 +229,21 @@ NORETURN void sched_task_exit(void)
     sched_active_thread = NULL;
     cpu_switch_context_exit();
 }
+
+#ifdef MODULE_CORE_PRIORITY_INHERITANCE
+void sched_change_priority(thread_t *thread, uint8_t priority)
+{
+    assert(thread && (priority < SCHED_PRIO_LEVELS));
+
+    if (thread->priority == priority) {
```

Comment:
This will never happen for the mutex but will for the msg. Maybe you can add an `if` in the msg to reduce the impact in the mutex.